scico/test/test_loss.py

```diff
@@ -0,0 +1,131 @@
+import numpy as np
+
+from jax.config import config
+
+import pytest
+
+# enable 64-bit mode for output dtype checks
+config.update("jax_enable_x64", True)
+
+
+from prox import prox_test
+
+import scico.numpy as snp
+from scico import functional, linop, loss
+from scico.random import randn
+
+
+class TestLoss:
+    def setup_method(self):
+        n = 4
+        dtype = np.float64
+        A, key = randn((n, n), key=None, dtype=dtype, seed=1234)
+        D, key = randn((n,), key=key, dtype=dtype)
+        W, key = randn((n,), key=key, dtype=dtype)
+        W = 0.1 * W + 1.0
+        self.Ao = linop.MatrixOperator(A)
+        self.Ao_abs = linop.MatrixOperator(snp.abs(A))
+        self.Do = linop.Diagonal(D)
+        self.W = linop.Diagonal(W)
+        self.y, key = randn((n,), key=key, dtype=dtype)
+        self.v, key = randn((n,), key=key, dtype=dtype)  # point for prox eval
+        scalar, key = randn((1,), key=key, dtype=dtype)
+        self.scalar = scalar.copy().ravel()[0]
+
+    def test_generic_squared_l2(self):
+        A = linop.Identity(input_shape=self.y.shape)
+        l = functional.SquaredL2Norm()
+        L0 = loss.Loss(self.y, A=A, l=l, scale=0.5)
+        L1 = loss.SquaredL2Loss(y=self.y, A=A)
```

Comment:
perhaps call then `f`, `f0`, `f1`? We use `L` for the Lipschitz constant or upper case for operators.

Reply:
Agreed, but `L` is already widely used in the same way in all the other existing tests in the same file. I'm inclined to leave addressing this for a separate PR.